creating_images/guidelines.adoc

```diff
@@ -299,44 +299,27 @@ privileged user.
 Because the user ID of the container is generated dynamically, it will not have
 an associated entry in *_/etc/passwd_*. This can cause problems for applications
 that expect to be able to look up their user ID. One way to address this problem
-is to use link:https://cwrap.org/nss_wrapper.html[nss wrapper] and dynamically
-create a *_passwd_* file with the container's user ID as part of the image's
-start script:
+is to dynamically create a *_passwd_* file with the container's user ID as part
+of the image's `ENTRYPOINT` script. This what a Dockerfile might include:
 
 ----
-export USER_ID=$(id -u)
-export GROUP_ID=$(id -g)
-envsubst < ${HOME}/passwd.template > /tmp/passwd
-export LD_PRELOAD=/usr/lib64/libnss_wrapper.so
-export NSS_WRAPPER_PASSWD=/tmp/passwd
-export NSS_WRAPPER_GROUP=/etc/group
+ENV USER_NAME=nobody \
+    USER_UID=99
+RUN chown -R ${USER_UID}:0 /some/directory \
+    && chmod -R g=u /some/directory /etc/passwd
+RUN sed "s@${USER_NAME}:x:${USER_UID}:@${USER_NAME}:x:\${USER_ID}:@g" /etc/passwd > /etc/passwd.template
+ENTRYPOINT [ "uid_entrypoint" ]
+USER 99
 ----
 
-Where *_passwd.template_* contains:
+Where *_uid_entrypoint_* contains:
 
 ----
-root:x:0:0:root:/root:/bin/bash
-bin:x:1:1:bin:/bin:/sbin/nologin
-daemon:x:2:2:daemon:/sbin:/sbin/nologin
-adm:x:3:4:adm:/var/adm:/sbin/nologin
-lp:x:4:7:lp:/var/spool/lpd:/sbin/nologin
-sync:x:5:0:sync:/sbin:/bin/sync
-shutdown:x:6:0:shutdown:/sbin:/sbin/shutdown
-halt:x:7:0:halt:/sbin:/sbin/halt
-mail:x:8:12:mail:/var/spool/mail:/sbin/nologin
-operator:x:11:0:operator:/root:/sbin/nologin
-games:x:12:100:games:/usr/games:/sbin/nologin
-ftp:x:14:50:FTP User:/var/ftp:/sbin/nologin
-nobody:x:99:99:Nobody:/:/sbin/nologin
-postgres:x:${USER_ID}:${GROUP_ID}:PostgreSQL Server:${HOME}:/bin/bash
-----
-
-Additionally, you must install the *nss_wrapper* and *gettext* packages in your
-image for this to work. The latter provides the `envsubst` command. For
-example you can add this line to your *_Dockerfile_* for yum-based  images:
-
-----
-RUN yum -y install nss_wrapper gettext
+if [ -w /etc/passwd ]; then
+  if ! whoami &> /dev/null; then
+    sed "s@${USER_NAME}:x:\${USER_ID}:@${USER_NAME}:x:$(id -u):@g" /etc/passwd.template > /etc/passwd
```

Comment:
if you look at what i linked, in our jenkins image we just append a new line to the /etc/passwd file for the current container uid.  We skip doing this if and only if the container is being run as the jenkins user (997), but I think you can leave that check out for purposes of this example.  In fact that check isn't even correct since the jenkins user is 998 (the group is 997).